src/Map.ts

```diff
@@ -0,0 +1,825 @@
+  foldMapWithKey(O)(M)(fa, (_, a) => f(a))
+
+/**
+ * @since 1.14.0
+ */
+const foldr = <K>(O: Ord<K>): (<A, B>(fa: Map<K, A>, b: B, f: (a: A, b: B) => B) => B) => {
+  return (fa, b, f) => foldrWithKey(O)(fa, b, (_, a, b) => f(a, b))
+}
+
+/**
+ * @since 1.14.0
+ */
+const reduceWithKey = <K>(O: Ord<K>): (<A, B>(fa: Map<K, A>, b: B, f: (k: K, b: B, a: A) => B) => B) => {
+  const keysO = keys(O)
+  return <A, B>(fa: Map<K, A>, b: B, f: (k: K, b: B, a: A) => B): B => {
+    let out: B = b
+    const ks = keysO(fa)
+    const len = ks.length
+    for (let i = 0; i < len; i++) {
+      const k = ks[i]
+      out = f(k, out, fa.get(k)!)
+    }
+    return out
+  }
+}
+
+/**
+ * @since 1.14.0
+ */
+const foldMapWithKey = <K>(O: Ord<K>): (<M>(M: Monoid<M>) => <A>(fa: Map<K, A>, f: (k: K, a: A) => M) => M) => {
+  const keysO = keys(O)
+  return <M>(M: Monoid<M>) => <A>(fa: Map<K, A>, f: (k: K, a: A) => M): M => {
+    let out: M = M.empty
+    const ks = keysO(fa)
+    const len = ks.length
+    for (let i = 0; i < len; i++) {
+      const k = ks[i]
+      out = M.concat(out, f(k, fa.get(k)!))
+    }
+    return out
+  }
+}
+
+/**
+ * @since 1.14.0
+ */
+const foldrWithKey = <K>(O: Ord<K>): (<A, B>(fa: Map<K, A>, b: B, f: (k: K, a: A, b: B) => B) => B) => {
+  const keysO = keys(O)
+  return <A, B>(fa: Map<K, A>, b: B, f: (k: K, a: A, b: B) => B): B => {
+    let out: B = b
+    const ks = keysO(fa)
+    const len = ks.length
+    for (let i = len - 1; i >= 0; i--) {
+      const k = ks[i]
+      out = f(k, fa.get(k)!, out)
+    }
+    return out
+  }
+}
+
+/**
+ * Create a Map with one key/value pair
+ *
+ * @since 1.14.0
+ */
+export const singleton = <K, A>(k: K, a: A): Map<K, A> => {
+  return new Map<K, A>([[k, a]])
+}
+
+/**
+ * @since 1.14.0
+ */
+function traverseWithKey<F extends URIS3>(
+  F: Applicative3<F>
+): <K, U, L, A, B>(ta: Map<K, A>, f: (k: K, a: A) => Type3<F, U, L, B>) => Type3<F, U, L, Map<K, B>>
+function traverseWithKey<F extends URIS2>(
+  F: Applicative2<F>
+): <K, L, A, B>(ta: Map<K, A>, f: (k: K, a: A) => Type2<F, L, B>) => Type2<F, L, Map<K, B>>
+function traverseWithKey<F extends URIS>(
+  F: Applicative1<F>
+): <K, A, B>(ta: Map<K, A>, f: (k: K, a: A) => Type<F, B>) => Type<F, Map<K, B>>
+function traverseWithKey<F>(
+  F: Applicative<F>
+): <K, A, B>(ta: Map<K, A>, f: (k: K, a: A) => HKT<F, B>) => HKT<F, Map<K, B>>
+function traverseWithKey<F>(
+  F: Applicative<F>
+): <K, A, B>(ta: Map<K, A>, f: (k: K, a: A) => HKT<F, B>) => HKT<F, Map<K, B>> {
+  return <K, A, B>(ta: Map<K, A>, f: (k: K, a: A) => HKT<F, B>) => {
+    let fm: HKT<F, Map<K, B>> = F.of(empty)
+    const entries = ta.entries()
+    let e: IteratorResult<[K, A]>
+    while (!(e = entries.next()).done) {
+      const [key, a] = e.value
+      fm = F.ap(F.map(fm, m => (b: B) => new Map(m).set(key, b)), f(key, a))
+    }
+    return fm
+  }
+}
+
+/**
+ * @since 1.14.0
+ */
+function traverse<F extends URIS3>(
```

Comment:
overloadings can be removed

Reply:
Same as before

Reply:
I made this change. Marking as resolved.